### ISSUE TITLE
Support more collection type conversions from input list

### ISSUE THREAD

**Issue by gao-artur** [graphql-dotnet/graphql-dotnet#3930]
[ObjectExtensions.GetPropertyValue](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Extensions/ObjectExtensions.cs#L225) current behaviour:

When the `fieldType` is `ISet<T>`, `IReadOnlySet<T>`, `IImmutableSet<T>`, or any of the concrete implementations it returns `List<T>`. I believe it should be considered a bug.

For `ImmutableArray<T>` it throws
> System.NotSupportedException
> Specified method is not supported.
>    at System.Collections.Immutable.ImmutableArray`1.System.Collections.IList.Add(Object value)

For `ImmutableList<T>` it throws
> System.MissingMethodException
> No parameterless constructor defined for type 'System.Collections.Immutable.ImmutableList`1[System.Double]'.

**Comment by gao-artur** [comment]
In v8.0 it will be possible to workaround with the new `Parse` method, but I think the support for these types should be built-in.

**Comment by gao-artur** [comment]
> Supporting HashSet will lose the order of the items and may cause input errors (validation errors) to occur due to duplicate entries

I think we should dedup input items by using `HashSet<>.Add()` method. The question about the comparer brings us back to the discussion about overriding default values on input objects:
```c#
public class MyInput
{
    public HashSet<string> MySet { get; } = new HashSet<string>(StringComparer.InvariantCultureIgnoreCase);
}
```
 The user may want to init the property and expect the created HashSet to be filled instead of overriding it (people asked MS to implement this behavior in M.E.Configuration, and they did in .NET8)

> Do we support Queue or Stack? ... What about the Frozen classes?

IMO, all these types shouldn't be used in input objects. Frozen types are very expensive to construct and should be used if not changed for a long time. It's not a good fit for an input object of a single request. Queue and Stack fill like a "business logic types" rather than data carriers for a request. I'd expect users to contract them manually when needed.

> Another possibility: what if we update/copy the ValueConverter class to support list types?

I like this most. The `ValueConverter` was the first thing I checked to register the conversion, but unfortunately, I found collections are not supported.
